File: leetCode/412fizzBuzz/solution.java

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> strings = new ArrayList<>();
        for(int i=1;i<=n;i++){
            if(i%15 == 0){
                strings.add("FizzBuzz");
            }else
            if(i%3 == 0){
                strings.add("Fizz");
            }else
            if(i%5 == 0){
                strings.add("Buzz");
            }else{
                // strings.add("" + i);
                strings.add(Integer.toString(i));
            }
            
        }
        return strings;
    }
}

File: leetCode/412fizzBuzz/solution.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    var ans = [];
    for(var i = 1;i<=n;i++){
        switch((i%3 == 0? 1 : 0) + (i%5 == 0? 2 : 0)){
            case 3: 
                ans.push("FizzBuzz");
                break;
            case 2: 
                ans.push("Buzz");
                break;
            case 1: 
                ans.push("Fizz");
                break;
            default:
                ans.push(''+ i);
        }
    }
    return ans;
};

console.log(fizzBuzz(15));